docs/tutorial/data-flow.md

```diff
@@ -0,0 +1,153 @@
+## Data Flow
+
+There are several methods for sending data between controllers and components. This tutorial provides an overview of the different possibilities and best practices for your application.
+
+## From parents to children
+
+### Parameters
+
+The simplest method is using the `@Param` annotation, which can be employed when displaying a controller with the `show()` or `initAndRender()` methods. This annotation can also bind various properties, making the data flow responsive to changes.
+
+```java
+@Component
+public class MyComponent {
+
+  @Param("data")
+  Data data;
+
+  // ...
+}
+```
+
+```java
+@Controller
+public class MyController {
+
+  @OnRender
+  void createSubs() {
+    MyComponent component = app.initAndRender(new MyComponent(), Map.of("data", myData));
+    // ...
+  }
+
+}
+```
+
+### Java Constructs
+
+Since controllers and components are Java objects, basic concepts like constructor parameters, setters, and getters can also be used. After creating a component instance, methods can be called on it for passing data or other effects.
```

Comment:
Should specify that unlike parameters, this can be done at any point even after initialization.